app/models/review.rb

```diff
@@ -2,23 +2,87 @@ class Review < ApplicationRecord
   include WithScoring
 
   belongs_to :assigner,       class_name: 'User',
-                              foreign_key: 'created_id'
+                              foreign_key: 'assigner_id'
   belongs_to :reviewer,       class_name: 'User',
                               foreign_key: 'reviewer_id'
   belongs_to :submission,     class_name: 'GrantSubmission::Submission',
                               foreign_key: 'grant_submission_submission_id',
                               counter_cache: true,
                               inverse_of: :reviews
-  has_many :criteria_reviews
+  has_one :grant,             through: :submission
+  has_many :grant_criteria,   through: :grant,
+                              source: :criteria
+
+  has_many :criteria_reviews, dependent: :destroy
   has_many :criteria,         through: :criteria_reviews
 
-  # calculating averages on the fly will be easier.
-  # consider whether averages should be cached.
+  accepts_nested_attributes_for :criteria_reviews
+
+  validates_presence_of     :reviewer
+  validates_presence_of     :overall_impact_score, unless: :new_record?
```

Comment:
Should this be here? What if someone is only asked to comment on one of the other dependent criteria. Is that a further in the future add?

Reply:
Opened #133 about this.